### PR TITLE
fixes issues/96

### PR DESCRIPTION
**Pull request by simouel**
pull request with the code from issue #96

### CODE REVIEW

File: Classes/Service/CompatibilityService.php [hunk 1]
```diff
@@ -35,7 +35,8 @@ public function isBackend(): bool
             // Backwards compatibility: for TYPO3 versions lower than 11.0
             return TYPO3_MODE === 'BE';
         } else {
-            return ApplicationType::fromRequest($GLOBALS['TYPO3_REQUEST'])->isBackend();
+            return ($GLOBALS['TYPO3_REQUEST'] ?? null) instanceof ServerRequestInterface && ApplicationType::fromRequest($GLOBALS['TYPO3_REQUEST'])->isBackend();
```
[the remark this answers is not in the document]
Comment (IchHabRecht) [reply Classes/Service/CompatibilityService.php:38]: BTW: no need for that else part after a return in an if clause

**Review by megamisan — CHANGES_REQUESTED** [review]
Could you include the required use line in the patch ?

`use Psr\Http\Message\ServerRequestInterface;`

**Comment by weakbit** [comment]
@simouel the code in the master branch changed already and returns false for frontend AND backend in CLI mode as it is neighter of it.
Also the code using that function is no need to execute in CLI.
Thank you.

File: Classes/Service/CompatibilityService.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Moongazer) [reply Classes/Service/CompatibilityService.php:38]: I run in this issue and modified it like this which works:
```
    public function isBackend(): bool
    {
        if (version_compare(VersionNumberUtility::getNumericTypo3Version(), '11.0.0') === -1) {
            // Backwards compatibility: for TYPO3 versions lower than 11.0
            return TYPO3_MODE === 'BE';
        }

        return empty($GLOBALS['TYPO3_REQUEST']) || ApplicationType::fromRequest($GLOBALS['TYPO3_REQUEST'])->isBackend();
    }
```

**Comment by Moongazer** [comment]
@weakbit The current master contains multiple fixes already, wouldn't it time for a release, please? The last one was ~1 year ago

**Pull request closed without merging**